src/ralph/backends/http/async_lrs.py

```diff
@@ -0,0 +1,361 @@
+            ignore_errors (bool): If `True`, errors during the write operation
+                are ignored and logged. If `False` (default), a `BackendException`
+                is raised if an error occurs.
+            operation_type (OperationType or None): The mode of the write operation.
+                If `operation_type` is `None`, the `default_operation_type` is used
+                instead. See `OperationType`.
+            simultaneous (bool): If `True`, chunks requests will be made concurrently.
+                If `False` (default), chunks will be sent sequentially
+            max_num_simultaneous (int or None): If simultaneous is `True`, the maximum
+                number of chunks to POST concurrently. If `None` (default), no limit is
+                set.
+        """
+        data = iter(data)
+        try:
+            first_record = next(data)
+        except StopIteration:
+            logger.info("Data Iterator is empty; skipping write to target.")
+            return 0
+
+        if not operation_type:
+            operation_type = self.default_operation_type
+
+        if operation_type in (
+            OperationType.APPEND,
+            OperationType.UPDATE,
+            OperationType.DELETE,
+        ):
+            msg = f"{operation_type.value} operation type is not supported."
+            logger.error(msg)
+            raise BackendParameterException(msg)
+
+        if not target:
+            target = self.statements_endpoint
+
+        target = ParseResult(
+            scheme=urlparse(self.base_url).scheme,
+            netloc=urlparse(self.base_url).netloc,
+            path=target,
+            query="",
+            params="",
+            fragment="",
+        ).geturl()
+
+        if (max_num_simultaneous is not None) and max_num_simultaneous < 1:
+            msg = "max_num_simultaneous must be a stricly positive integer"
+            logger.error(msg)
+            raise BackendParameterException(msg)
+
+        if (not simultaneous) and (max_num_simultaneous is not None):
+            msg = "max_num_simultaneous can only be used with `simultaneous=True`"
+            logger.error(msg)
+            raise BackendParameterException(msg)
+
+        # Gather only one chunk at a time when not using the simultaneous option
+        if not simultaneous:
+            max_num_simultaneous = 1
+
+        data = chain((first_record,), data)
+
+        logger.debug(
+            "Start writing to the %s endpoint (chunk size: %s)", target, chunk_size
+        )
+
+        # Create tasks
+        tasks = set()
+        for chunk in chunked(data, chunk_size):
+            tasks.add(self._post_and_raise_for_status(target, chunk, ignore_errors))
+
+        # Run POST tasks
+        result = await self._gather_with_limited_concurrency(
+            max_num_simultaneous, *tasks
+        )
+        statements_count = sum(result)
+
+        logger.info("Posted %d statements", statements_count)
```

Comment:
I think that logging (at level `INFO`) the number of posted statements could be the responsibility of the caller (either CLI or LRS). We could still log this at level `DEBUG`. WDYT?

Reply:
I agree !